dsgrid/config/dataset_config.py

```diff
@@ -198,6 +213,15 @@ class DatasetConfigModel(DSGBaseModel):
         ),
         # TODO: it would be nice to extend the description here with a CLI example of how to list the project's data source IDs.
     )
+    data_schema_type: DataSchemaType = Field(
+        title="data_schema_type",
+        description="Discriminator for data schema",
+        options=DataSchemaType.format_for_docs(),
+    )
+    data_schema: Union[StandardDataSchemaModel, OneTableDataSchemaModel] = Field(
```

Comment:
I have experienced weird problems with Pydantic and Union types. Since you are using `data_schema_type` as a discriminator and creating the object yourself, it would be fine to leave this defined as
```
data_schema: Any
```
If everything is working then you don't need to change it.

Reply:
The union is working for me